auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -192,6 +196,10 @@ public Void then(@NonNull Task<GoogleApiClient> task) throws Exception {
         // Facebook sign out
         LoginManager.getInstance().logOut();
 
+        // Twitter sign out
+        if (!Fabric.isInitialized()) TwitterProvider.initialize(activity);
+        Twitter.logOut();
```

Comment:
@samtstern I thought so too, but `Twitter.logOut()` already does that for us:
```java
// In Twitter.java
public static void logOut() {
    checkInitialized();
    getInstance().core.logOut(); // core == TwitterCore
}

// In TwitterCore.java
public void logOut() {
    checkInitialized();
    final SessionManager<TwitterSession> sessionManager = getSessionManager();
    if (sessionManager != null) {
        sessionManager.clearActiveSession();
    }
}

public static SessionManager<TwitterSession> getSessionManager() {
    checkInitialized();
    return getInstance().core.getSessionManager(); // Same reference as logOut in TwitterCore.java
}
```

Actually, now that I'm paying more attention, it looks like that's all `Twitter#logOut()` does.

The docs for `TwitterCore#logOut()` say it logs the user out, so I think that's all we have to do:
```java
/**
 * Logs out the user, clearing user session. This will not make a network request to invalidate
 * the session.
 *
 * @throws java.lang.IllegalStateException if {@link io.fabric.sdk.android.Fabric}
 *          or {@link TwitterCore} has not been initialized.
 */
```